Extend the method section with text processors (#35)
Ctrl-Alt-Defeat-Foundation-Models/informatiCup2024Paper: paper.tex

\section{Method}
To test and evaluate the effect of different data augmentation techniques on the detection capabilities of fake image or text detectors, we created a pipeline of three components: The first component is a generator, which is a model capable of generating fake images or texts. The second component is a processor which augments the data through a specified method, after given a generated image or text. The last component is a detector, which suggests an input being either real or fake. This way we can compare it to the ground truth, to evaluate the capabilities of the detector. By comparing the results with augmented artefacts and with the generated ones without augmentation, we can evaluate the effectiveness of different AI detector-tricking techniques.

> We used the aforementioned models for generating and detecting texts and images. As for image processors, one was already given in the description of the "InformatiCup2024" \cite{InformatiCup}, the image noising using the Gaussian noise. Image noise is the random variation of brightness or color information in images produced by the sensor and circuitry of a scanner or digital camera. The reason why the Gaussian one, also called normal noise, is being used, is because it is being caused by natural sources and the most common noise that affects images naturally \cite{mohammed2016study}.

Reviewer: ```suggestion
We used the aforementioned models for generating and detecting texts and images. As for image processors, one was already given in the description of the "InformatiCup2024" \cite{InformatiCup}, the image noising using the Gaussian noise. Image noise is the random variation of brightness or color information in images produced by the sensor and circuitry of a scanner or digital camera. We used gaussian noice, because it can occure due to natural sources and it is the most common noise that affects images naturally. \cite{mohammed2016study}
```

Author: I disagree here. I do not want to say that Gaussian noise is "good", but rather because it is common to find.

Reviewer 2: I also wouldn't change it

Reviewer: now @EhrenbruderRoschtbraten ?

There are several types of noise that can affect images. Some of these noise models are: Gaussian noise, White noise, Fractal noise, Salt \& Pepper noise, Periodic noise, Quantization noise, Speckle noise, Poisson noise, Poisson-Gaussian noise, Structured noise, Gamma noise and Rayleigh noise \cite{gonzalez2009digital}. The three common types of image noise are: Gaussian noise, Salt \& Pepper noise, and Speckle noise \cite{al2010comparative}. That is why we used those three, as well as the Poisson noise.

> As for text processors, we evaluated three different methods of augmenting generated texts. The first approach was to replace every white-space in a given text by two white-spaces. The second approach was to add some typos to the texts. This has been implemented based on the nlpaug library, which is capable of simulating certain typos as if they were made on a keyboard by a human. The typos include mistakes, like hitting a wrong key, which is close to the right one. This results in typos often created by humans. The last approach is a translator, which has been used to translate the given text into a certain language (in this case French) and re-translate it into english. As translator are not deterministic but aim to keep the semantic meaning, the resulting text should be semantically the same, but might have a different formulation.

Reviewer: ```suggestion
As for text processors, we evaluated three different methods of augmenting generated texts. The first approach was to replace every white-space in a given text by two white-spaces. The second approach was to add some typos to the texts. This has been implemented based on the nlpaug library, which is capable of simulating certain typos as if they were made on a keyboard by a human. The typos include mistakes, like hitting a wrong key, which is close to the right one. This results in typos often created by humans. The last approach is a translator, which has been used to translate the given text into a certain language (in this case French) and re-translate it into english. As translators are not deterministic but aim to keep the semantic meaning, the resulting text should be semantically the same, but might have a different formulation.
```

Author: I disagree partially here, the typo is fine, but French should be uppercase, because in this case it means the name of the language (which is uppercase) and not the adjective.

Reviewer 2: I agree with @1Maxnet1 here

> As for text processors, we evaluated on three different methods of augmenting the text. The first approach is to replace every white-space in a given text, by two white-spaces. The second approach is to add some typos to the texts. This has been implemented based on the nlpaug library, which is capable of simulating certain typos as if they were made on a keyboard by a human. The typos include mistakes, like not hitting the right key, but a key close to the one, that should have been hit instead. This results in typos close to real typos by humans. The last approach is a translator, which has been used to translate the given text into a certain language (in this case French) and re-translate it into english. As translator are not deterministic but aim to keep the semantic meaning, the resulting text should be semantically the same, but might have a different formulation.

Reviewer 2: ```suggestion
As for text processors, we evaluated three different methods of augmenting generated texts. The first approach was to replace every white-space in a given text by two white-spaces. The second approach was to add some typos to the texts. This has been implemented based on the nlpaug library, which is capable of simulating certain typos as if they were made on a keyboard by a human. The typos include mistakes, like hitting a wrong key, which is close to the right one. This results in typos often created by humans. The last approach is a translator, which has been used to translate the given text into a certain language (in this case French) and re-translate it into english. As translator are not deterministic but aim to keep the semantic meaning, the resulting text should be semantically the same, but might have a different formulation.
```
